File: ts-sdk/src/sdk/models/operations/getarticles.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetArticlesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=collection_id" })
  collectionId?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=state" })
  state?: shared.PerPageParam30to1000Enum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tag" })
  tag?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tags" })
  tags?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tags_exclude" })
  tagsExclude?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=top" })
  top?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=username" })
  username?: string;
}


export class GetArticlesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetArticlesQueryParams;
}


export class GetArticlesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.ArticleIndex })
  articleIndices?: shared.ArticleIndex[];

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getarticles.go
package operations

import (
	"github.com/speakeasy-sdks/forem-sdks/go-client-sdk/pkg/models/shared"
)

type GetArticlesQueryParams struct {
	CollectionID *int32                           `queryParam:"style=form,explode=true,name=collection_id"`
	Page         *int32                           `queryParam:"style=form,explode=true,name=page"`
	PerPage      *int32                           `queryParam:"style=form,explode=true,name=per_page"`
	State        *shared.PerPageParam30to1000Enum `queryParam:"style=form,explode=true,name=state"`
	Tag          *string                          `queryParam:"style=form,explode=true,name=tag"`
	Tags         *string                          `queryParam:"style=form,explode=true,name=tags"`
	TagsExclude  *string                          `queryParam:"style=form,explode=true,name=tags_exclude"`
	Top          *int32                           `queryParam:"style=form,explode=true,name=top"`
	Username     *string                          `queryParam:"style=form,explode=true,name=username"`
}

type GetArticlesRequest struct {
	QueryParams GetArticlesQueryParams
}

type GetArticlesResponse struct {
	ArticleIndices []shared.ArticleIndex
	ContentType    string
	StatusCode     int64
}
